src/ralph/backends/data/swift.py

```diff
@@ -0,0 +1,380 @@
+                instead. See `BaseOperationType`.
+
+        Returns:
+            int: The number of written records.
+
+        Raises:
+            BackendException: If a failure during the write operation occurs and
+                `ignore_errors` is set to `False`.
+            BackendParameterException: If a backend argument value is not valid.
+        """
+        try:
+            first_record = next(iter(data))
+        except StopIteration:
+            logger.info("Data Iterator is empty; skipping write to target.")
+            return 0
+        if not operation_type:
+            operation_type = self.default_operation_type
+
+        if not target:
+            target = f"{self.default_container}/{now()}-{uuid4()}"
+            logger.info(
+                (
+                    "Target not specified; using default container "
+                    "with random object name: %s"
+                ),
+                target,
+            )
+        elif "/" not in target:
+            target = f"{self.default_container}/{target}"
+            logger.info(
+                "Container not specified; using default container: %s",
+                self.default_container,
+            )
+
+        target_container, target_object = target.split("/", 1)
+
+        if operation_type in [
+            BaseOperationType.APPEND,
+            BaseOperationType.DELETE,
+            BaseOperationType.UPDATE,
+        ]:
+            msg = "%s operation_type is not allowed."
+            logger.error(msg, operation_type.name)
+            if not ignore_errors:
+                raise BackendParameterException(msg % operation_type.name)
+
+        if operation_type in [BaseOperationType.CREATE, BaseOperationType.INDEX]:
+            if target_object in list(self.list(target=target_container)):
+                msg = "%s already exists and overwrite is not allowed for operation %s"
+                logger.error(msg, target_object, operation_type)
+                if not ignore_errors:
+                    raise BackendException(msg % (target_object, operation_type))
+
+            if isinstance(first_record, dict):
+                data = [
+                    json.dumps(statement).encode(self.locale_encoding)
+                    for statement in data
+                ]
+
+            try:
+                self.connection.put_object(
+                    container=target_container, obj=target_object, contents=data
+                )
+                resp = self.connection.head_object(
+                    container=target_container, obj=target_object
+                )
+            except ClientException as err:
+                msg = "Failed to write to object %s: %s"
+                error = err.msg
+                logger.error(msg, target_object, error)
+                if not ignore_errors:
+                    raise BackendException(msg % (target_object, error)) from err
+
+        count = sum(1 for _ in data)
```

Comment:
Why not `len(data)` ?

Reply:
As the type of `data` is `Iterable`, it does not have to implement the `__len__` method, I can't see another way than to sum all iterations